### PR TITLE
fix:The slash of a string cannot be removed using json_serialize_to_string

### PR DESCRIPTION
**Pull request by ych988**
fix:IF the put request's json payload is string type, The slash of a string cannot be removed using json_serialize_to_string.
example: {"command": "open"}   → {\\"command\\": \\"open\\"}  → \open\

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/device-sdk-c/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] I am not introducing a breaking change (if you are, flag in conventional commit message with `BREAKING CHANGE:` describing the break)
- [ ] I am not introducing a new dependency (add notes below if you are)
- [ ] I have added unit tests for the new feature or bug fix (if not, why?)
- I have fully tested (add details below) this the new feature or bug fix (if not, why?)
- [ ] I have opened a PR for the related docs change (if not, why?)
  <link to docs PR>

### CODE REVIEW

**Comment by lenny-intel** [comment]
@ych988 , We require a BUG issue to be create for all bug prior to submitting fixes.
Please submit a bug issue here: https://github.com/edgexfoundry/device-sdk-c/issues

**Comment by ych988** [comment]
#492 has been created.

**Comment by cloudxxx8** [comment]
@FelixTing please help review this PR when you have time

File: src/c/reqdata.c [hunk 1]
```diff
@@ -104,7 +104,13 @@ extern const char *edgex_reqdata_get (const edgex_reqdata_t *data, const char *n
   {
     JSON_Value *value = json_object_get_value(data->json, name);
     if (value) {
-      result = json_serialize_to_string(value);
+      if(json_value_get_type(value) == JSONString){
+        //The slash of a string cannot be removed using json_serialize_to_string
+        result = json_object_get_string(data->json, name);
```
Comment (FelixTing) [inline src/c/reqdata.c:109]:
```suggestion
        result = json_value_get_string(value);
```

**Comment by FelixTing** [comment]
@ych988 Please follow the [Committing Code Guidelines](https://wiki.edgexfoundry.org/display/FA/Committing+Code+Guidelines#CommittingCodeGuidelines-GitHubFlow) to create separate branches and PRs for the commits https://github.com/edgexfoundry/device-sdk-c/pull/490/commits/5a81276ddb39c68d2fe0e953fa52f6adfeb106a9 and https://github.com/edgexfoundry/device-sdk-c/pull/490/commits/97ed87bd81bbe0e316ce27e281fc1129d7b47373.

**Comment by ych988** [comment]
create new request for the commits 5a81276 and 97ed87b

**Pull request closed without merging**